hyperspy/_signals/hologram_image.py

```diff
@@ -0,0 +1,303 @@
+# -*- coding: utf-8 -*-
+# Copyright 2007-2016 The HyperSpy developers
+#
+# This file is part of  HyperSpy.
+#
+#  HyperSpy is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+#  HyperSpy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with  HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+import numpy as np
+from skimage import draw
+from scipy.fftpack import ifft2, fftshift, fft2
+from numpy.linalg import norm
+import matplotlib.pyplot as plt
+import matplotlib.cm as cm
+from hyperspy._signals.signal2d import Signal2D
+
+
+class HologramImage(Signal2D):
+    """Image subclass for holograms acquired via electron holography."""
+
+    _signal_type = 'hologram'
+
+    def reconstruct_phase(self, reference=None, rec_param=None, show_phase=False, return_param=False,
+                          fresnel_fringe_filter=False, verbose=False):
+        """Reconstruct holography data
+
+        Parameters
+        ----------
+        reference : ndarray, class:`~hyperspy._signals.Image`, `~hyperspy._signals.HologramImage`
+            Vacuum reference hologram.
+        rec_param : tuple
+            Reconstruction parameters in sequence (SBrect(x0, y0, x1, y1), SB size)
+        show_phase : boolean
+            set True to plot phase after the reconstruction
+        return_param : boolean
+            set True to return reconstruction as tuple, useful for batch processing
+        fresnel_fringe_filter : boolean
+            set True to apply an automatic filtering of biprisms fresnel fringe
+        verbose : boolean
+            set True to see details of the reconstruction (i.e. SB selection)
+
+        Returns
+        -------
+        wave : WaveImage
+            Reconstructed electron wave. By default object wave is devided by reference wave
+        rec_param : tuple
+            see the description in Parameters (optional, triggered by return_param flag)
+
+        Notes
+        -----
+        The reconstruction parameters assigned interactively if rec_param is not givven
+
+        See Also
+        --------
+        _reconstruct
+
+        """
+
+        (sx, sy) = self.data.shape
```

Comment:
In accordance with later comments, I personally would use
```python
holo_size = self.data.shape
```
and always refer the tuples indices instead of having `sx` and `sy` separately, but that's just my opinion. If you want to stay with your convention, just adapt the later comments accordingly.